pallets/connectors/src/message.rs

```diff
@@ -118,6 +110,10 @@ impl<
 				message.append(&mut token_name.encode());
 				message.append(&mut token_symbol.encode());
 
+				let mut encoded_price = price.encode();
+				encoded_price.reverse();
+				message.append(&mut encoded_price);
```

Comment:
What about implementing a custom encoding for a `Price` type and here simply using `.encode()`?

Reply:
Or something similar for types that needs a particular encoding, using a wrapper type. Maybe not now, but for the feature, if it grows, it could be easy to make mistakes

Reply:
> What about implementing a custom encoding for a Price type and here simply using .encode()?

May be a good idea. But I guess that would mess up with decoding the current data on storage? What I tried doing, actually, was having a bound on the type, `Into<u128>`, to which I could convert to and then call `.to_be_bytes()`. The problem is that this approach needs to cascade down to the `pallet_connectors` Config types (`Balance`, `Rate`) as well as the `PoolInspect::PoolId` and `PoolInspect::TrancheId`; that's why I didn't go after this approach.

> Or something similar for types that needs a particular encoding, using a wrapper type. Maybe not now, but for the feature, if it grows, it could be easy to make mistakes

I'm curious! can you expand?

Reply:
Yeah, I was thinking in something like this:
```rust
#[derive(Decode, Clone, PartialEq, TypeInfo, Encode, Decode)]
#[cfg_attr(feature = "std", derive(Debug))]
pub enum Message<Domain, PoolId, TrancheId, Balance, Rate> {
    Invalid,
    AddPool {
        pool_id: XcmPoolId,
    },
    AddTranche {
        pool_id: XcmPoolId,
        tranche_id: XcmTrancheId,
        token_name: XcmName<TOKEN_NAME_SIZE>,
        token_symbol: XcmName<TOKEN_SYMBOL_SIZE>,
    },
    //...
}
// NOTE: No longer need an custom `encode()` method for `Message`.
```
And then, each XcmPoolId has a custom encoding for its special encoding:
```rust
struct XcmPoolId(PoolId);
impl Encode for XcmPoolId {
    fn encode(&self) -> Vec<u8> {
        let mut encoded_pool_id = self.0.encode();
        encoded_pool_id.reverse();
        encoded_pool_id
    }
}
// Same for price
```

Reply:
Thanks for expanding on this! Note that this are not "xcm" types, this is encoding requirements specific to Connectors. And if that's ok for you I'd rather leave this as is and consider revisit it when we work on Connectors v2. How does that sound?

Reply:
@lemunozm I just created a simple `to_be` function that does this. This seems to be the cleanest, least invasive, and more straightforward way to improve this a bit to wrap up this PR.

Reply:
I like much more the `to_be()` factorization 🙌🏻 . 
Following your approach, I was wondering if something like this could work, following your approach:
```rust
trait EncodeBigEndian: Encode {
    fn encode_be(&self) -> Vec<u8> {
        let encoded = self.encode();
        encoded.reverse();
        encoded
    }
}
impl<T: Encode> EncodeBigEndian for T {}
```
Usage:
```rust
price.encode_be()
```
Anyways, I'm ok as it is. 👍🏻